index.js

```diff
@@ -50,103 +50,100 @@ Limiter.prototype.inspect = function () {
  * that expire after N seconds:
  *
  *  - limit:<id>:count
- *  - limit:<id>:limit
- *  - limit:<id>:reset
  *
  * @param {Function} fn
  * @api public
  */
 
 Limiter.prototype.get = function (fn) {
   var count = this.prefix + 'count';
-  var limit = this.prefix + 'limit';
-  var reset = this.prefix + 'reset';
   var duration = this.duration;
   var max = this.max;
   var db = this.db;
 
   function create() {
-    var ex = (Date.now() + duration) / 1000 | 0;
-
     db.multi()
-      .set([count, max, 'PX', duration, 'NX'])
-      .set([limit, max, 'PX', duration, 'NX'])
-      .set([reset, ex, 'PX', duration, 'NX'])
+      .set(count, max, 'PX', duration, 'NX')
       .exec(function (err, res) {
         if (err) return fn(err);
 
         // If the request has failed, it means the values already
         // exist in which case we need to get the latest values.
-        if (isFirstReplyNull(res)) return mget();
-
-        fn(null, {
-          total: max,
-          remaining: max,
-          reset: ex
+        if (isReplyNull(res)) return get();
+
+        db.pttl(count, function(err, data) {
+          if(err) return fn(err);
+          fn(null, {
+            total: max,
+            remaining: max,
+            // Adding Date.now to ensure change is non-breaking. Can be removed for more accuracy
+            // later. reset should also be in milliseconds (unlike now) for more accuracy
+            reset: (~~data + Date.now()) / 1000
+          });
         });
       });
   }
 
   function decr(res) {
-    var n = ~~res[0];
-    var max = ~~res[1];
```

Comment:
Isn't this an issue? When a new max is set, shouldn't the keys be updated? I think the best solution would be to do a check like this,

`var n = ~~res[0];

if(n > max) resetTheKeyWithNewMax();`

Like you said, here n would be 98 and max would be 10. So the key would reset with value as 9 instead of 98.